### PR TITLE
add a tool to compute the volume integral of a plotfile field

### PR DESCRIPTION
**Pull request by zingale**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Tools/Plotfile/fvolumesum.cpp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                                        // Cartesian
+                                        for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+                                            vol *= dx[idim];
+                                        }
+                                    } else if (coord == 1) {
+                                        // axisymmetric V = pi (r_r**2 - r_l**2) * dz
+                                        //                = pi dr * dz * (r_r + r_l)
+                                        //                = 2 pi r dr dz
+                                        vol = 2 * M_PI * p[0] * dx[0] * dx[1];
+                                    }
```
Comment (WeiqunZhang) [inline Tools/Plotfile/fvolumesum.cpp:179]: There is no `coord == 2` here.  But it is supported for `lev < fine_level`.

Comment (zingale) [reply Tools/Plotfile/fvolumesum.cpp:179]: good catch.  Fixed.

**Review by WeiqunZhang — APPROVED** [review]

**Comment by philip-blakely** [comment]
Hello; I've just tried using the fvolumesum tool for checking conservation in our AMReX-based code.

Very useful, but I've found a small error: you need to initialise Real vol=1; (in two places).
Otherwise the final integral is undefined (often zero) in the Cartesian case.
Also, fvolumesum ideally needs to be added to the GNUMakefile.

**Comment by zingale** [comment]
you are right -- I had implemented it and tested it for axisymmetric, but the vol initialization for Cartesian was wrong.  That is fixed in #2596

**Pull request merged**